csharp/test/Apache.Arrow.Tests/ArrayBuilderTests.cs

```diff
@@ -38,6 +39,92 @@ public void PrimitiveArrayBuildersProduceExpectedArray()
             TestArrayBuilder<DoubleArray, DoubleArray.Builder>(x => x.Append(10).Append(20).Append(30));
         }
 
+
+        [Fact]
+        public void ListArrayBuilder()
+        {
+            var stringField = new Field("item", StringType.Default, true);
+            var listBuilder = new ListArray.Builder(new ListType(stringField));
```

Comment:
Is it confusing that the `ListArray.Builder` constructor takes a `ListType`? Would it make more sense for it to take the `ValueDataType`? It is fairly obvious that I am going to be building a "list" if I'm using a `ListArray.Builder`, so why make the user have to build a `ListType` up to describe it?

If we take the `ValueDataType`, I think it would be easier to write:

```C#
var listBuilder = new ListArray.Builder(StringType.Default);
```

vs. the above:

```C#
var stringField = new Field("item", StringType.Default, true);
var listBuilder = new ListArray.Builder(new ListType(stringField));
```

Reply:
```c#
var listBuilder = new ListArray.Builder(StringType.Default);
```

API will be usable for many users but we can't remove

```c#
var stringField = new Field("item", StringType.Default, true);
var listBuilder = new ListArray.Builder(new ListType(stringField));
```

API. Because `ListType`'s  Apache Arrow spec can have a field name as optional value. It means that advanced users may want to specify `ListType`'s field name.

So how about adding `new ListArray.Builder(StringType.Default)` API instead of replacing `new ListArray.Builder(new ListType(stringField))` API with `new ListArray.Builder(StringType.Default)` API?

Reply:
> So how about adding `new ListArray.Builder(StringType.Default)` API instead of replacing `new ListArray.Builder(new ListType(stringField))` API with `new ListArray.Builder(StringType.Default)` API?

Is it confusing when creating a nested `ListType` array?
When specifing `ListType` to the `ListArray.Builder` argument , some user may expect the child type is `ListType`.
I suppose it is better the parameter is unified to either `ListType` only or `ValueDataType` only.
How do you feel about it?

> Because ListType's Apache Arrow spec can have a field name as optional value. It means that advanced users may want to specify ListType's field name.

It seems enough to be able to specify the child field of `ListType`. (Please tell me if I'm wrong)
So how about adding these two constructors if we change this constructor?

```
Builder(IArrowType valueDataType){...}
Builder(Field childField){...}
```

Reply:
Ah, sorry. You're right.

```c#
Builder(IArrowType valueDataType){...}
Builder(Field childField){...}
```

will be better.

Reply:
Thank you.

I added these constructors.
I left the original `Builder(ListType dataType)` as `internal` because the added constructors can be simplified by using the original constructor and ArrowArrayBuilderFactory.Build should call the original constructor.

```cs
Builder(IArrowType valueDataType)  : this(new ListType(valueDataType)) {...}
Builder(Field valueField) : this(new ListType(valueField)) {...}
```